lib/spack/spack/test/cmd/env.py

```diff
@@ -547,6 +547,55 @@ def test_with_config_bad_include(env_deactivate, capfd):
     assert 'no/such/file.yaml' in err
 
 
+def test_env_with_include_config_files_same_basename():
+    test_config = """\
+        env:
+            include:
+                - ./path/to/included-config.yaml
+                - ./second/path/to/include-config.yaml
+            specs:
+                [libelf, mpileaks]
+            """
+
+    _env_create('test', StringIO(test_config))
+    e = ev.read('test')
+
+    fs.mkdirp(os.path.join(e.path, 'path', 'to'))
+    with open(os.path.join(
+            e.path,
+            './path/to/included-config.yaml'), 'w') as f:
+        f.write("""\
+        packages:
+          libelf:
+              version: [0.8.10]
+        """)
+
+    fs.mkdirp(os.path.join(e.path, 'second', 'path', 'to'))
+    with open(os.path.join(
+            e.path,
+            './second/path/to/include-config.yaml'), 'w') as f:
+        f.write("""\
+        packages:
+          mpileaks:
+              version: [2.2]
+        """)
+
+    with e:
+        e.concretize()
+
+    mpileaks_found = False
+    libelf_found = False
+
+    for x in e._get_environment_specs():
+        if mpileaks_found and libelf_found:
+            break
+        if x.satisfies('mpileaks@2.2'):
+            mpileaks_found = True
+        elif x.satisfies('libelf@0.8.10'):
+            libelf_found = True
+    assert(mpileaks_found and libelf_found)
```

Comment:
I think the assertion logic at the end of the test could be cleaned up. Something like
```suggestion
    for expected in ('mpileaks@2.2', 'libelf@0.8.10'):
        assert any(cspec.satisfies(expected) for _, cspec in e.concretized_specs())
```

Another option would be to add the `recurse_dependencies=False` option to `_get_environment_specs`, that way there wouldn't be any additional specs to iterate over.

Reply:
The test as written could also technically pass for the wrong reason. `mpileaks` depends on `libelf`, so if the `mpileaks` dependency on `libelf` used version 0.8.10 but the top-level spec did not, the test should fail but this version would pass because `mpileaks` is first in the spec list.

Reply:
I'm going with the `recurse_dependencies=False`.